detox/local-cli/utils/cliErrorHandling.js

```diff
@@ -0,0 +1,15 @@
+const loggedErrors = new WeakSet();
+
+function markErrorAsLogged(error) {
+  loggedErrors.add(error);
+  return error;
+}
+
+function isErrorAlreadyLogged(error) {
+  return loggedErrors.has(error);
+}
+
+module.exports = {
+  markErrorAsLogged,
+  isErrorAlreadyLogged,
+};
```

Comment:
So do I totally dislike this solution? Of course not! But do I _like_ this solution? Hmm........ 🤔
It _is_ kind of elegant... and as for concerns about garbage collection not kicking in on time - experimentation shows we're on the good side of things (yes, I checked). So I guess the real question is, does the error really have to be reported via [`logger.error.end()`](https://github.com/wix/Detox/pull/3749/files#diff-548b574af3df69470e3a3a4c901c845b4180a406c29b1d621ec1eaa25ec93255R145) in TestRunnerCommand? Meaning, to some degree, this duplication situation is self-inflicted, isn't it?

Reply:
I can't come up with a more creative solution right now. It makes sense to call `log.*end()` there because there might be a few times we call an underlying test runner command and we want to log each command separately. Maybe we'll revisit this later with some additions/rewrites to the `local-cli/*`.